variables.tf

```
@@ -46,6 +46,26 @@ variable "burstable_mode" {
   default     = null
 }
 
+variable "market_type" {
+  type        = string
+  description = "(Optional) Type of market for the instance. Valid values are `spot` and `capacity-block`. If the value set is `spot`, the `spot_options_attributes` variable becomes adjustable."
```

The `terraform-docs` automatically marks inputs as `optional` or `required` based on if there is a `default` set.

```suggestion
  description = "Type of market for the instance. Valid values are `spot` and `capacity-block`. If the value set is `spot`, the `spot_options_attributes` variable becomes adjustable."
```